### PR TITLE
feat: qbd direct export settings testing with api changes

### PR DESCRIPTION
**Pull request by DhaaraniCIT**
### Description
qbd direct export settings testing with api changes
Screenshot 2024-10-16 at 5 48 28 PM

Screenshot 2024-10-16 at 5 48 54 PM

## Clickup

https://app.clickup.com/t/86cwpmtxe

### CODE REVIEW

File: src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.ts [hunk 1]
```diff
@@ -168,7 +167,7 @@ export class QbdDirectExportSettingsComponent implements OnInit{
 
   employeeMappingWatcher() {
     this.exportSettingsForm.controls.reimbursableExportType.valueChanges.subscribe((reimbursableExportTypeValue) => {
-      if (reimbursableExportTypeValue === QbdDirectReimbursableExpensesObject.BILL) {
+      if (reimbursableExportTypeValue === QbdDirectReimbursableExpensesObject.BILL || reimbursableExportTypeValue === QbdDirectReimbursableExpensesObject.JOURNAL_ENTRY) {
```
Comment (ashwin1111) [inline src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.ts:170]: no need to add `reimbursableExportTypeValue === QbdDirectReimbursableExpensesObject.JOURNAL_ENTRY` condition, for JE we should get input from user

Comment (DhaaraniCIT) [reply src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.ts:170]: yes, for a fresh user when we select JE, it throws an error and the following fields are not being displayed in UI, to fix this I did this, tried defaulting value in form initialization but the form has null values for all, could not find where it is becoming null, so to fix this I did this

Comment (ashwin1111) [reply src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.ts:170]: let's debug over a call and close it - we shouldn't default it

Comment (ashwin1111) [reply src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.ts:170]: pls block some time for tomo

**Pull request merged**